brave/brave-core
# Adjust sidebar colors

fix https://github.com/brave/brave-browser/issues/21590

So far, sidebar can't aligned with custom theme
because sidebar only knows brave's dark/light theme color.

With below changes, sidebar colors are aligned well with custom theme also.

Used toolbar color as a sidebar background color.
Used toolbar button's hover color for sidebar item/arrow button hover color.
Picked proper color between light/dark color for builtin item icon color based on toolbar color.
Used toolbar area separator color(separator between toolbar and web contents) for sidebar border.

![image](https://user-images.githubusercontent.com/6786187/158109675-4f238154-46f9-46ff-8b60-93c48574791e.png)  ![image](https://user-images.githubusercontent.com/6786187/158109734-297c4b3e-bbce-4f8c-8539-3e0c040f5db4.png) ![image](https://user-images.githubusercontent.com/6786187/158109875-dd4f2b6c-7219-4bcc-a19a-3e2477f9c895.png) ![image](https://user-images.githubusercontent.com/6786187/158109995-5da524d8-b3f8-4ca1-a41b-857abb061ac9.png)

![image](https://user-images.githubusercontent.com/6786187/158112596-d2868624-d9f9-4b1e-aab3-bfa1bb64fe62.png)
![image](https://user-images.githubusercontent.com/6786187/158112695-eeb7a961-8542-458e-adab-cf410eb433a5.png)
![image](https://user-images.githubusercontent.com/6786187/158112835-a3fb6ecf-caa7-41ce-9a51-3379bdcb95a1.png)
![image](https://user-images.githubusercontent.com/6786187/158109566-bcde4840-d98e-4a6b-b81e-94aec3b30039.png)




Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch brave
2. Load brave://settings/manageProfile and try any theme color
3. Check sidebar colors are aligned with new them like above screenshot
